src/axom/inlet/tests/inlet_function.cpp

```diff
@@ -321,6 +321,273 @@ TEST(inlet_function, simple_vec3_to_vec3_array_of_struct)
   EXPECT_FLOAT_EQ(second_result[2], 18);
 }
 
+TEST(inlet_function, dimension_dependent_result)
+{
+  std::string testString =
+    "foo = function (v) "
+    "first = 2 * v.x "
+    "last = 2 * v.y "
+    "if v.dim == 2 then "
+    "return Vector.new(first, last) "
+    "else "
+    "return Vector.new(first, 0, last) "
+    "end "
+    "end";
+  DataStore ds;
+  auto inlet = createBasicInlet(&ds, testString);
+
+  inlet.addFunction("foo",
+                    FunctionTag::Vector,
+                    {FunctionTag::Vector},
+                    "foo's description");
+
+  auto callable =
+    inlet["foo"].get<std::function<FunctionType::Vector(FunctionType::Vector)>>();
+
+  FunctionType::Vector input_3d({3.5, 0.5, 7.5});
+  auto result = callable(input_3d);
+
+  EXPECT_EQ(result.dim, 3);
+  EXPECT_FLOAT_EQ(result.vec[0], 7);
+  EXPECT_FLOAT_EQ(result.vec[1], 0);
+  EXPECT_FLOAT_EQ(result.vec[2], 1);
+
+  FunctionType::Vector input_2d({3.5, 0.5});
+  result = callable(input_2d);
+
+  EXPECT_EQ(result.dim, 2);
+  EXPECT_FLOAT_EQ(result.vec[0], 7);
+  EXPECT_FLOAT_EQ(result.vec[1], 1);
+}
+
+template <typename Ret, typename... Args>
+Ret checkedCall(const sol::protected_function& func, Args&&... args)
+{
+  auto tentative_result = func(std::forward<Args>(args)...);
+  EXPECT_TRUE(tentative_result.valid());
+  return tentative_result;
+}
+
+TEST(inlet_function, lua_usertype_basic)
```

Comment:
Suggestion: This test warrants some comments since it is different in nature than the previous ones that use the tagged types

Reply:
Good feedback, I've moved these into a separate `inlet_function_usertype` suite and added a comment explaining their purpose